app/bundles/CoreBundle/Security/Permissions/CorePermissions.php

```diff
@@ -215,7 +215,8 @@ public function generatePermissions(array $permissions)
                 $entity->setName(strtolower($name));
 
                 $bit   = 0;
-                $class = $this->getPermissionObject($bundle);
+                $pluginBundles = $this->getPluginBundles();
```

Comment:
Can you move this to be outside the foreach loops? Somewhere around line 207 or 208. Not sure it needs to be fetched on each foreach iteration.